### PR TITLE
Create windows.md

### PR DESCRIPTION
**Pull request by ihuibin**
Signed-off-by: bin.hui <bin.hui@daocloud.io>

### CODE REVIEW

File: content/zh/docs/tasks/debug/debug-cluster/windows.md [hunk 1]
```diff
@@ -0,0 +1,303 @@
...
+
+   Ensure that your Pod is getting scheduled to a [compatable](https://docs.microsoft.com/virtualization/windowscontainers/deploy-containers/version-compatibility) Windows Node.
+
+   More information on how to specify a compatable node for your Pod can be found in [this guide](docs/setup/production-environment/windows/user-guide-windows-containers/#ensuring-os-specific-workloads-land-on-the-appropriate-container-host).
+-->
+2. 我的 pod 状态显示 'ErrImgPull' 或者 ‘ImagePullBackOff’
+
+   保证你的 Pod 在 Windows 节点的兼容范围 [compatable](https://docs.microsoft.com/virtualization/windowscontainers/deploy-containers/version-compatibility) 。
+
+   关于如何为你的 Pod 指定一个兼容节点的更多信息可以查看这个指南。 [this guide](docs/setup/production-environment/windows/user-guide-windows-containers/#ensuring-os-specific-workloads-land-on-the-appropriate-container-host)
```
Comment (Sea-n) [inline content/zh/docs/tasks/debug/debug-cluster/windows.md:51]:
```suggestion
   关于如何为你的 Pod 指定一个兼容节点的更多信息可以查看这个指南。[這份指南](/zh/docs/setup/production-environment/windows/user-guide-windows-containers/#ensuring-os-specific-workloads-land-on-the-appropriate-container-host)
```

File: content/zh/docs/tasks/debug/debug-cluster/windows.md [hunk 2]
```diff
@@ -0,0 +1,303 @@
...
+   "ExceptionList": [
+                   "10.244.0.0/16",  # Cluster subnet
+                   "10.96.0.0/12",   # Service subnet
+                   "10.127.130.0/24" # Management (host) subnet
+               ]
+   ```
+-->
+2. 我的 Windows Pod 不能 ping 通外界资源
+
+   Windows Pod 没有为ICMP协议编写出站规则。 但是，TCP/UDP 是支持的。 当试图演示与集群外部资源的连接时,  可以把 `ping <IP>` 替换为 `curl <IP>` 命令。
```
Comment (Sea-n) [inline content/zh/docs/tasks/debug/debug-cluster/windows.md:99]:
```suggestion
   Windows Pod 没有为 ICMP 协议编写出站规则，但 TCP/UDP 是支持的。当试图演示与集群外部资源的连接时，可以把 `ping <IP>` 替换为 `curl <IP>` 命令。
```

File: content/zh/docs/tasks/debug/debug-cluster/windows.md [hunk 3]
```diff
@@ -0,0 +1,303 @@
...
+
+   如果你仍然面临问题， 最可能的是你需要额外关注
+   [cni.conf](https://github.com/Microsoft/SDN/blob/master/Kubernetes/flannel/l2bridge/cni/config/cni.conf)
+   的配置。你可以随时编辑这个静态文件。更新配置将应用于新的 Kubernetes 资源。
+
+   Kubernetes 的网络需求之一 (查看 [Kubernetes model](zh/docs/concepts/cluster-administration/networking/)) 是在内部不需要NAT的情况下进行集群通信。
+   为了遵守这一要求， 对于你不希望发生出站NAT的通信，这里有一个
+   [ExceptionList](https://github.com/Microsoft/SDN/blob/master/Kubernetes/flannel/l2bridge/cni/config/cni.conf#L20) 。
+   然而，这也意味着你需要从 'ExceptionList' 中排除你试图查询的外部IP。只有这样，流量才会来自你的 Windows Pod 被正确地 SNAT 以接收来自外部环境的响应。 在这方面，你的 ' cni.conf ' 中的 ' ExceptionList ' 应该如下所示:
```
Comment (Sea-n) [inline content/zh/docs/tasks/debug/debug-cluster/windows.md:108]:
```suggestion
   然而，这也意味着你需要从 `ExceptionList` 中排除你试图查询的外部 IP。只有这样，流量才会来自你的 Windows Pod 被正确地 SNAT 以接收来自外部环境的响应。 在这方面，你的 `cni.conf` 中的 `ExceptionList` 应该如下所示：
```

File: content/zh/docs/tasks/debug/debug-cluster/windows.md [hunk 4]
```diff
@@ -0,0 +1,303 @@
...
+   C:\k\kube-proxy.exe --hostname-override=$(hostname)
+   ```
+-->
+3. 我的 Windows 节点无法访问 ' NodePort ' 类型服务
+
+   从节点本身访问本地 NodePort 失败，是一个已知的限制。 可以从其他节点或外部客户端正常访问 NodePort 。
+
+4. 容器的 vnic 和 HNS endpoints 正在被删除
+
+   这个问题可能是当' hostname-override '参数没有传递给 [kube-proxy] (/docs/reference/command-line-tools-reference/kube-proxy/)时引起的。想要解决这个问题，用户需要将主机名传递给 kube-proxy ，如下所示：
```
Comment (Sea-n) [inline content/zh/docs/tasks/debug/debug-cluster/windows.md:139]: Fix this too.

**Review by Sea-n — COMMENTED** [review]
Please review after machine translation before submit your PR.

Netlify preview: https://deploy-preview-34006--kubernetes-io-main-staging.netlify.app/zh/docs/tasks/debug/debug-cluster/windows

File: content/zh/docs/tasks/debug/debug-cluster/windows.md [hunk 5]
```diff
@@ -0,0 +1,303 @@
...
+   If you would like to build your own pause infrastructure container be sure to include [wincat](https://github.com/kubernetes/kubernetes/tree/master/build/pause/windows/wincat).
+-->
+7. DNS 解析工作异常
+
+   在本节中 [section](#dns-limitations) 查看 Windows 系统的 DNS 限制 。
+
+8. `kubectl port-forward` 失败并且 "unable to do port forwarding: wincat not found"
+
+   这是在 Kubernetes 1.15 中通过包括 `wincat.exe` 在 pause 基础架构容器 `mcr.microsoft.com/oss/kubernetes/pause:3.6` 中实现的。
+   请确保使用 Kubernetes 支持的版本。如果你想构建自己的 pause 基础架构容器，请确保包含 [wincat](https://github.com/kubernetes/kubernetes/tree/master/build/pause/windows/wincat)。
```
Comment (tengqm) [inline content/zh/docs/tasks/debug/debug-cluster/windows.md:198]:
```suggestion
   请确保使用 Kubernetes 的受支持版本。如果你想构建自己的 pause 基础架构容器，
   请确保其中包含 [wincat](https://github.com/kubernetes/kubernetes/tree/master/build/pause/windows/wincat)。
```

File: content/zh/docs/tasks/debug/debug-cluster/windows.md [hunk 6]
```diff
@@ -0,0 +1,303 @@
...
+   If you are behind a proxy, the following PowerShell environment variables must be defined:
+
+   ```PowerShell
+   [Environment]::SetEnvironmentVariable("HTTP_PROXY", "http://proxy.example.com:80/", [EnvironmentVariableTarget]::Machine)
+   [Environment]::SetEnvironmentVariable("HTTPS_PROXY", "http://proxy.example.com:443/", [EnvironmentVariableTarget]::Machine)
+   ```
+-->
+9. 我的 Kubernetes 安装失败，因为我的 Windows 服务器节点使用了代理服务器
+
+   如果使用了代理服务器，必须要定义下面的 PowerShell 环境变量：
```
Comment (tengqm) [inline content/zh/docs/tasks/debug/debug-cluster/windows.md:211]:
```suggestion
   如果使用了代理服务器，必须定义下面的 PowerShell 环境变量：
```

File: content/zh/docs/tasks/debug/debug-cluster/windows.md [hunk 7]
```diff
@@ -0,0 +1,303 @@
...
+   Whenever a previously deleted node is being re-joined to the cluster, flannelD
+   tries to assign a new pod subnet to the node. Users should remove the old pod
+   subnet configuration files in the following paths:
+
+   ```powershell
+   Remove-Item C:\k\SourceVip.json
+   Remove-Item C:\k\SourceVipRequest.json
+   ```
+-->
+
```
Comment (tengqm) [inline content/zh/docs/tasks/debug/debug-cluster/windows.md:231]: ### Flannel 故障排查   {#flannel-troubleshooting}

File: content/zh/docs/tasks/debug/debug-cluster/windows.md [hunk 8]
```diff
@@ -0,0 +1,303 @@
...
+1. 使用 Flannel 时，我的节点在重新加入集群后出现问题
+
+   当先前删除的节点重新加入集群时, flannelD 尝试为节点分配一个新的 pod 子网。 用户应该在以下路径中删除旧的pod子网配置文件：
```
Comment (tengqm) [inline content/zh/docs/tasks/debug/debug-cluster/windows.md:234]:
```suggestion
   当先前删除的节点重新加入集群时, flannelD 尝试为节点分配一个新的 Pod 子网。
   用户应该在以下路径中删除旧的 Pod 子网配置文件：
```

File: content/zh/docs/tasks/debug/debug-cluster/windows.md [hunk 9]
```diff
@@ -0,0 +1,303 @@
...
+   A workaround is to relaunch `start.ps1` or relaunch it manually as follows:
+
+   ```powershell
+   [Environment]::SetEnvironmentVariable("NODE_NAME", "<Windows_Worker_Hostname>")
+   C:\flannel\flanneld.exe --kubeconfig-file=c:\k\config --iface=<Windows_Worker_Node_IP> --ip-masq=1 --kube-subnet-mgr=1
+   ```
+-->
+2. Flanneld 卡在 "Waiting for the Network to be created"
+
+   关于这个问题有很多报告 [issue](https://github.com/coreos/flannel/issues/1066)；
```
Comment (tengqm) [inline content/zh/docs/tasks/debug/debug-cluster/windows.md:254]:
```suggestion
   关于这个[问题](https://github.com/coreos/flannel/issues/1066)有很多报告 ；
```

File: content/zh/docs/tasks/debug/debug-cluster/windows.md [hunk 10]
```diff
@@ -0,0 +1,303 @@
...
+   很可能是设置 flannel 网络的管理IP的时机问题。
```
Comment (tengqm) [inline content/zh/docs/tasks/debug/debug-cluster/windows.md:255]:
```suggestion
   很可能是 flannel 网络管理 IP 的设置时机问题。
```

File: content/zh/docs/tasks/debug/debug-cluster/windows.md [hunk 11]
```diff
@@ -0,0 +1,303 @@
...
+   一个变通方法是重新启动 `start.ps1` 或手动通过以下方式解决：
```
Comment (tengqm) [inline content/zh/docs/tasks/debug/debug-cluster/windows.md:256]:
```suggestion
   一个变通方法是重新启动 `start.ps1` 或按如下方式手动重启：
```

File: content/zh/docs/tasks/debug/debug-cluster/windows.md [hunk 12]
```diff
@@ -0,0 +1,303 @@
...
+
+   ```powershell
+   [Environment]::SetEnvironmentVariable("NODE_NAME", "<Windows_Worker_Hostname>")
+   C:\flannel\flanneld.exe --kubeconfig-file=c:\k\config --iface=<Windows_Worker_Node_IP> --ip-masq=1 --kube-subnet-mgr=1
```
Comment (tengqm) [inline content/zh/docs/tasks/debug/debug-cluster/windows.md:260]:
```suggestion
   C:\flannel\flanneld.exe --kubeconfig-file=c:\k\config --iface=<Windows 工作节点 IP> --ip-masq=1 --kube-subnet-mgr=1
```

File: content/zh/docs/tasks/debug/debug-cluster/windows.md [hunk 13]
```diff
@@ -0,0 +1,303 @@
...
+   ```
+-->
+2. Flanneld 卡在 "Waiting for the Network to be created"
+
+   关于这个问题有很多报告 [issue](https://github.com/coreos/flannel/issues/1066)；
+   很可能是设置 flannel 网络的管理IP的时机问题。
+   一个变通方法是重新启动 `start.ps1` 或手动通过以下方式解决：
+
+   ```powershell
+   [Environment]::SetEnvironmentVariable("NODE_NAME", "<Windows_Worker_Hostname>")
```
Comment (tengqm) [inline content/zh/docs/tasks/debug/debug-cluster/windows.md:259]:
```suggestion
   [Environment]::SetEnvironmentVariable("NODE_NAME", "<Windows 工作节点主机名>")
```

File: content/zh/docs/tasks/debug/debug-cluster/windows.md [hunk 14]
```diff
@@ -0,0 +1,303 @@
...
+   ```env
+   FLANNEL_NETWORK=10.244.0.0/16
+   FLANNEL_SUBNET=10.244.4.1/24
+   FLANNEL_MTU=1500
+   FLANNEL_IPMASQ=true
+   ```
+-->
+3. 我的 Windows Pod 无法启动，因为缺少 `/run/flannel/subnet.env`
+
+   这表明 Flannel 没有正确启动。 你可以尝试重启`flanneld.exe` 或者你可以从 Kubernetes master 的 `/run/flannel/subnet.env` 手动拷贝到 Windows worker node `C:\run\flannel\subnet.env` 并且修改 `FLANNEL_SUBNET` 那一行为不同数值。 例如，节点子网为 10.244.4.1/24：
```
Comment (tengqm) [inline content/zh/docs/tasks/debug/debug-cluster/windows.md:280]:
```suggestion
   这表明 Flannel 没有正确启动。你可以尝试重启`flanneld.exe` 或者你可以将 Kubernetes 控制节点的
   `/run/flannel/subnet.env` 文件手动拷贝到 Windows 工作节点上，放在 `C:\run\flannel\subnet.env`；
   并且将 `FLANNEL_SUBNET` 行修改为不同取值。例如，如果期望节点子网为 10.244.4.1/24：
```

File: content/zh/docs/tasks/debug/debug-cluster/windows.md [hunk 15]
```diff
@@ -0,0 +1,303 @@
...
+<!-- 
+### Further investigation
+
+If these steps don't resolve your problem, you can get help running Windows containers on Windows nodes in Kubernetes through:
+
+* StackOverflow [Windows Server Container](https://stackoverflow.com/questions/tagged/windows-server-container) topic
+* Kubernetes Official Forum [discuss.kubernetes.io](https://discuss.kubernetes.io/)
+* Kubernetes Slack [#SIG-Windows Channel](https://kubernetes.slack.com/messages/sig-windows)
+-->
+### 进一步研究
```
Comment (tengqm) [inline content/zh/docs/tasks/debug/debug-cluster/windows.md:297]:
```suggestion
### 进一步探查   {#further-investigation}
```

File: content/zh/docs/tasks/debug/debug-cluster/windows.md [hunk 16]
```diff
@@ -0,0 +1,303 @@
...
+
+如果这些步骤都不能解决你的问题，在 Kubernetes 中使用 Windows 节点上运行 Windows 容器你可以通过以下方式获取帮助：
```
Comment (tengqm) [inline content/zh/docs/tasks/debug/debug-cluster/windows.md:299]:
```suggestion
如果这些步骤都不能解决你的问题，你可以通过以下方式获得关于在 Kubernetes 中运行 Windows 容器的帮助：
```

File: content/zh/docs/tasks/debug/debug-cluster/windows.md [hunk 17]
```diff
@@ -0,0 +1,303 @@
...
+
+* StackOverflow [Windows Server Container](https://stackoverflow.com/questions/tagged/windows-server-container) topic
+* Kubernetes Official Forum [discuss.kubernetes.io](https://discuss.kubernetes.io/)
```
Comment (tengqm) [inline content/zh/docs/tasks/debug/debug-cluster/windows.md:302]:
```suggestion
* Kubernetes 官方论坛 [discuss.kubernetes.io](https://discuss.kubernetes.io/)
```

**Comment by tengqm** [comment]
This is crazy. We have 35 commits for this PR now, and I have got 34 email notifications.
@ihuibin, can you do these commits at your local machine and then `git commit --amend` followed by a `git push -f`?

File: content/zh/docs/tasks/debug/debug-cluster/windows.md (hunk 4 above)
Comment (ihuibin) [reply content/zh/docs/tasks/debug/debug-cluster/windows.md]: OK

File: content/zh/docs/tasks/debug/debug-cluster/windows.md [hunk 18]
```diff
@@ -0,0 +1,303 @@
+---
+title: Windows 调试小技巧
+content_type: concept
+---
+<!--
+---
```
[the remark this answers is not in the document]
Comment (ihuibin) [reply content/zh/docs/tasks/debug/debug-cluster/windows.md]: OK

File: content/zh/docs/tasks/debug/debug-cluster/windows.md [hunk 19]
```diff
@@ -0,0 +1,303 @@
...
+title: Windows debugging tips
+content_type: concept
+---
```
[the remark this answers is not in the document]
Comment (ihuibin) [reply content/zh/docs/tasks/debug/debug-cluster/windows.md]: done

**Comment by ihuibin** [comment]
sorry，sorry

**Comment by ydFu** [comment]
> sorry，sorry

If you have problems with the way you use git commands, you can ask your questions. Otherwise, similar contributions will trouble the zh code view team.

**Comment by tengqm** [comment]
@ihuibin Interested in moving this forward (squash your commits)?

**Comment by ihuibin** [comment]
I squash 34 commits in github desktop.

**Comment by tengqm** [comment]
@ihuibin Please `git push -f` after you have squashed your commits.

**Comment by tengqm** [comment]
I'm kicking this in and leave improvements as future works.
/lgtm
/approve

**Pull request merged**